firebase-perf/src/main/java/com/google/firebase/perf/injection/modules/FirebasePerformanceModule.java

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.injection.modules;
+
+import androidx.annotation.NonNull;
+import com.google.android.datatransport.TransportFactory;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.inject.Provider;
+import com.google.firebase.installations.FirebaseInstallationsApi;
+import com.google.firebase.perf.config.ConfigResolver;
+import com.google.firebase.perf.internal.GaugeManager;
+import com.google.firebase.perf.internal.RemoteConfigManager;
+import com.google.firebase.remoteconfig.RemoteConfigComponent;
+import dagger.Module;
+import dagger.Provides;
+
+/**
+ * Provider for FirebasePerformance
+ *
+ * @hide
+ */
+@Module
+public class FirebasePerformanceModule {
+  private final FirebaseApp firebaseApp;
+  private final FirebaseInstallationsApi firebaseInstallations;
+  private final Provider<RemoteConfigComponent> remoteConfigComponentProvider;
+  private final Provider<TransportFactory> transportFactoryProvider;
+
+  //  private final RemoteConfigManager remoteConfigManager;
+  //  private final ConfigResolver configResolver;
+  //  private final GaugeManager gaugeManager;
+
+  public FirebasePerformanceModule(
+      @NonNull FirebaseApp firebaseApp,
+      @NonNull FirebaseInstallationsApi firebaseInstallations,
+      @NonNull Provider<RemoteConfigComponent> remoteConfigComponentProvider,
+      @NonNull Provider<TransportFactory> transportFactoryProvider) {
+    this.firebaseApp = firebaseApp;
+    this.firebaseInstallations = firebaseInstallations;
+    this.remoteConfigComponentProvider = remoteConfigComponentProvider;
+    this.transportFactoryProvider = transportFactoryProvider;
+
+    //    this.remoteConfigManager = RemoteConfigManager.getInstance();
+    //    this.configResolver = ConfigResolver.getInstance();
+    //    this.gaugeManager = GaugeManager.getInstance();
+  }
+
+  @Provides
+  FirebaseApp providesFirebaseApp() {
+    return firebaseApp;
+  }
```

Comment:
_Just for Info_ How are we using these APIs? Currently I don't see it being called from anywhere?

Reply:
Yes, it is not explicitly called anywhere. The method means that whenever we need a `FirebaseApp` type object, we will get `firebaseApp` as indicated in the methods body. The method name does not really matter. 

See https://dagger.dev/api/2.12/dagger/Provides.html. "The method's return type is bound to its returned value. The component implementation will pass dependencies to the method as parameters."